src/controller.hpp

```diff
@@ -23,6 +23,7 @@
 #include <functional>
 #include <memory>
 #include <string>
+#include <vector>
```

Comment:
While plausible that this may be needed, I did not see any compiler warning or error indicating that it is.